v2/handler.go

```diff
@@ -0,0 +1,73 @@
+package v2
+
+import (
+	"encoding/json"
+	"html/template"
+	"net/http"
+	"strings"
+)
+
+type checkHandler struct {
+	*HealthCheck
+}
+
+func Handler(hc *HealthCheck) func(w http.ResponseWriter, r *http.Request) {
+	ch := &checkHandler{hc}
+	return ch.handle
+}
+
+func (ch *checkHandler) handle(w http.ResponseWriter, r *http.Request) {
+	health := ch.health()
+
+	if strings.Contains(r.Header.Get("Accept"), "text/html") {
+		err := writeHTMLResp(w, health)
+		if err == nil {
+			return
+		}
+	}
+	w.Header().Set("Content-Type", "application/json")
+	enc := json.NewEncoder(w)
+	err := enc.Encode(health)
+	if err != nil {
+		panic("write this bit")
```

Comment:
"write this bit" is for sure just garbage, thanks for spotting this, and I think it shouldn't panic and crash the app, I think returning an 500 Internal Server Error with an explanation message would be what we want, because it's actually failing to encode the struct in to JSON.